src/SFML/System/Android/Activity.cpp

```diff
@@ -56,14 +57,24 @@ namespace sf
 {
 namespace priv
 {
-ActivityStates* getActivity(ActivityStates* initializedStates, bool reset)
+
+ActivityStates*& getActivityStatesPtr()
 {
     static ActivityStates* states = NULL;
+    return states;
+}
 
-    if (!states || reset)
-        states = initializedStates;
+void resetActivity(ActivityStates* initializedStates)
+{
+    getActivityStatesPtr() = initializedStates;
+}
 
-    return states;
+ActivityStates& getActivity()
+{
+    ActivityStates*& states = getActivityStatesPtr();
```

Comment:
Do you gain anything with a reference to a pointer?

Reply:
Not on the caller side, I will tweak this one.